boards/DCM/Src/App/App_Imu.c

```diff
@@ -0,0 +1,49 @@
+#include <assert.h>
+#include <stdlib.h>
+
+#include "App_Imu.h"
+
+struct Imu
+{
+    bool (*update_data)(void);
+    float (*get_acceleration_x)(void);
+    float (*get_acceleration_y)(void);
+    float (*get_acceleration_z)(void);
+};
+
+bool App_Imu_UpdateData(const struct Imu *imu)
```

Comment:
For the following functions I would add a const
```
bool App_Imu_UpdateData(const struct Imu * const imu)
```

Reply:
I changed these. For methods which take world as a parameter, should I update them to something like 
`App_Function(const struct DcmWorld * const world)`
as well?

Reply:
Only if you're getting something from the world and not modifying anything

Reply:
Oh whoops misread the comment. Yup you should do that but only in the .c file

Reply:
Why not the .h file? And how come I can do the double const thing even when I do things which modify world (such as modifying can_tx when I send a CAN message)

Reply:
Ignore this comment

> Only if you're getting something from the world and not modifying anything





Not sure if I've sent this to you but this should help clear things up : https://stackoverflow.com/questions/8808167/c-const-correctness-and-pointer-arguments

Reply:
The parameter passed to the SetPeriodicSignal is not a const struct atm